train.py

```
@@ -219,283 +270,312 @@
-            # Handle DDP model loading
-            target_model = model.module if hasattr(model, 'module') else model
+            # Handle DDP model loading (model might be DDP, checkpoint might be from DDP or single GPU)
+            # At this point, 'model' is the DDP-wrapped model if world_size > 1, or raw model if world_size == 1.
+            target_model_for_load = model.module if hasattr(model, 'module') else model
             model_state_dict = checkpoint["model_state_dict"]
 
-            # Handle module prefix mismatch
-            is_model_ddp = hasattr(model, 'module')
+            # Adjust keys if DDP state mismatch (e.g. loading DDP checkpoint into non-DDP model or vice-versa)
+            # This logic might need refinement if target_model_for_load is already DDP's module
+            is_target_model_ddp_module = not hasattr(model, 'module') # True if model is raw, or we already have model.module
```

Ruff (0.11.9)

Local variable `is_target_model_ddp_module` is assigned to but never used

Remove assignment to unused variable `is_target_model_ddp_module`

(F841)